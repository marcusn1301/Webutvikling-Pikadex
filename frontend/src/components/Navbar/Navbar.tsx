import { useEffect, useState } from "react";

import pokeballIcon from "../../assets/icons/pokeball_icon.svg";
import logo from "../../assets/Logo.svg";
import navbarEdge from "../../assets/pageEdges/navbarEdge.svg";

import leftEdge from "../../assets/pageEdges/leftEdge.svg";
import rightEdge from "../../assets/pageEdges/rightEdge.svg";

import pokeball from "../../assets/Pokeball.svg";
import { useScroll } from "../../hooks/useScroll";
import { HiMagnifyingGlass } from "react-icons/hi2";
import {
    ButtonsContainer,
    DropdownGridRow,
    DropdownHeader,
    DropdownOuter,
    FilterOuter,
    GridContainer,
    Logo,
    LogoContainer,
    LogoTextbox,
    NavbarEdge,
    NavbarOuter,
    PokeBall,
    PokeBallBtn,
    PokeBallBtnContainer,
    PokemonContainer,
    ResetBtn,
    SearchBtn,
    SearchOuter,
    SortByBox,
    SortTag,
    Tag,
    SelectedTags,
    TypesBox,
    OverflowWrapper,
    TagGrid,
    DropdownContainer,
    PokeballBoxSurpriseMe,
    PokeballBoxSurpriseMeText,
    RightEdge,
    LeftEdge,
    FilterTags,
    ButtonOuter,
} from "./NavbarStyles";

import pikachu from "../../assets/Navbar/pokemon/pikachu.svg";
import squirtle from "../../assets/Navbar/pokemon/squirtle.svg";
import charmander from "../../assets/Navbar/pokemon/charmander.svg";
import bulbasaur from "../../assets/Navbar/pokemon/bulbasaur.svg";

//Retrieved from https://gist.github.com/apaleslimghost/0d25ec801ca4fc43317bcff298af43c3
const list = [
    ["normal", "#A8A77A", "notPicked"],
    ["fire", "#EE8130", "notPicked"],
    ["water", "#6390F0", "notPicked"],
    ["electric", "#F7D02C", "notPicked"],
    ["grass", "#7AC74C", "notPicked"],
    ["ice", "#96D9D6", "notPicked"],
    ["fighting", "#C22E28", "notPicked"],
    ["poison", "#A33EA1", "notPicked"],
    ["ground", "#E2BF65", "notPicked"],
    ["flying", "#A98FF3", "notPicked"],
    ["psychic", "#F95587", "notPicked"],
    ["bug", "#A6B91A", "notPicked"],
    ["rock", "#B6A136", "notPicked"],
    ["ghost", "#735797", "notPicked"],
    ["dragon", "#6F35FC", "notPicked"],
    ["dark", "#705746", "notPicked"],
    ["steel", "#B7B7CE", "notPicked"],
    ["fairy", "#D685AD", "notPicked"],
];

interface NavbarI {
    searchTerm: string;
    setSearchTerm: React.Dispatch<React.SetStateAction<string>>;
    tags: any;
    setTags: React.Dispatch<React.SetStateAction<Array<Array<string>>>>;
    filterSearch: boolean;
    setFilterSearch: React.Dispatch<React.SetStateAction<boolean>>;
    surpriseMe: boolean;
    setSurpriseMe: React.Dispatch<React.SetStateAction<boolean>>;
    init: boolean;
    setInit: React.Dispatch<React.SetStateAction<boolean>>;
}

const Navbar = ({ searchTerm, setSearchTerm, tags, setTags, filterSearch, setFilterSearch, surpriseMe, setSurpriseMe, init, setInit }: NavbarI) => {
    const [showDropdown, setShowDropdown] = useState<boolean>(false);
    const [height, setHeight] = useState<string>("172px");
    const [update, setUpdate] = useState<boolean>(false);
    const [searchBarSearch, setSearchBarSearch] = useState("");
    const [colorList, setColorList] = useState<Array<Array<string>>>(list);

    let search = "";

    useEffect(() => {
        decideHeight();
    }, [showDropdown]);

    const decideHeight = () => {
        showDropdown ? setHeight("fit-content") : setHeight("172px");
    };

    const yOffsetThreshold = 70;

    const yOffset = useScroll();

    let small = false;

    let lineHeight: string = "70vh";
    let lineTop: string = "200px";
    let lineOffset: string = "1px";

    // If the user has scrolled over 70 px, increase the line heights
    if (yOffset > 70) {
        lineHeight = "80vh";
        lineTop = "120px";
        lineOffset = "5px";
    }

    if (yOffset > yOffsetThreshold) {
        small = true;
    }

    const checkHeight: () => string = () => {
        if (showDropdown) return "fit-content";
        else if (small) return "63px";
        else {
            return height;
        }
    };

    const handleSubmit = (e: any) => {
        e.preventDefault();
        setSearchTerm(searchBarSearch);
        setFilterSearch(!filterSearch);
        setShowDropdown(false);
    };

    const addTag = (tag: Array<string>, tagList: Array<Array<string>>, index: number) => {
        // If the tag is already in the list, dont add it
        if (tagList.includes(tag)) {
            return;
        }

        if (tagList.length >= 2) {
            colorList.map((color: Array<string>) => {
                if (color[0] === tagList[tagList.length - 1][1]) {
                    color[2] = "notPicked";
                    colorList[index][2] = "notPicked";
                }
            });

            tagList.pop();
            tagList.unshift(tag);
        } else {
            tagList.push(tag);
            colorList[index][2] = "picked";
        }

        // Sets the tags as state stored in Homepage.tsx component
        setTags(tagList);

        // The update state in only to the component to rerender
        setUpdate(!update);
    };

    const removeTag = (tagTerm: string, tagList: Array<Array<string>>) => {
        const filteredTags = tagList.filter((tag: Array<string>) => {
            return tag[0] !== tagTerm;
        });

        setTags(filteredTags);
        setUpdate(!update);

        colorList.map((color: Array<string>) => {
            if (color[0] === tagTerm) {
                color[2] = "notPicked";
            }
        });
    };

    const handleReset = () => {
        setTags([]);
        setSearchTerm("");
        setFilterSearch(!filterSearch);
        setSearchBarSearch("");
        setShowDropdown(false);
    };

    return (
        <>
            <NavbarOuter transition={small ? true : false} height={checkHeight()}>
                <OverflowWrapper
                    style={{
                        minHeight: small ? "63px" : "173px",
                        overflowY: !showDropdown ? "hidden" : "scroll",
                    }}
                >
                    <LogoContainer>
                        <LogoTextbox>
                            <Logo src={logo} />
                            <div>Pikadex</div>
                        </LogoTextbox>
                    </LogoContainer>
                    {!showDropdown && (
                        <img src={pikachu} alt="pikachu" className="pikachu" style={{ display: !small || showDropdown ? "block" : "none" }} />
                    )}
                    {!showDropdown && (
                        <PokemonContainer style={{ display: !small || showDropdown ? "block" : "none" }}>
                            <img src={pikachu} alt="pikachu" className="pikachu-small" />
                            <img src={bulbasaur} alt="bulbasaur" className="bulbasaur" />
                            <img src={charmander} alt="charizard" className="charmander" />
                            <img src={squirtle} alt="squirtle" className="squirtle" />
                        </PokemonContainer>
                    )}
                    <PokeBall src={pokeball} />
                    {small && !showDropdown ? (
                        ""
                    ) : (
                        <GridContainer>
                            {/* Grid row 1 */}
                            <SearchOuter>
                                <div className="filler"></div>
                                {/* Grid column 1 */}
                                <div className="searchContainer">
                                    <form onSubmit={handleSubmit}>
                                        <input
                                            type="text"
                                            placeholder="Search by name or index"
                                            className="searchbar"
                                            onChange={(e) => setSearchBarSearch(e.target.value)}
                                        />
                                    </form>
                                    <HiMagnifyingGlass size={25} className="searchIcon" onClick={handleSubmit} />
                                </div>
                                {/* Grid column 2 */}
                                <PokeBallBtnContainer
                                    onClick={() => {
                                        setSurpriseMe(!surpriseMe);
                                        setInit(false);
                                    }}
                                >
                                    <PokeballBoxSurpriseMe>
                                        <PokeballBoxSurpriseMeText>Surprise me!</PokeballBoxSurpriseMeText>
                                        <PokeBallBtn src={pokeballIcon} />
                                    </PokeballBoxSurpriseMe>
                                </PokeBallBtnContainer>
                            </SearchOuter>
                            {/* Grid row 2: selected type tags are displayed */}
                            <SelectedTags>
                                {/*Click on a tag in the filter and display them here*/}
                                {tags?.map((tag: string, index: number) => {
                                    return (
                                        <FilterTags background={tag[1]} key={index} onClick={() => removeTag(tag[0], tags)}>
                                            {tag[0]}
                                        </FilterTags>
                                    );
                                })}
                            </SelectedTags>
                        </GridContainer>
                    )}

                    {showDropdown && (
                        <>
                            <DropdownContainer>
                                <DropdownOuter>
                                    <DropdownGridRow>
                                        <DropdownHeader>Types</DropdownHeader>
                                        <TypesBox>
                                            <TagGrid>
                                                {/* Map over the list of tags with colors and text */}

                                                {colorList.map((item: Array<string>, index: number) => (
                                                    <Tag
                                                        backgroundColor={item[1]}
                                                        style={{ border: item[2] === "picked" ? "2px solid black" : "2px solid transparent" }}
                                                        key={index}
                                                        onClick={() => addTag(item, tags, index)}
                                                    >
                                                        {item[0]}
                                                    </Tag>
                                                ))}
                                            </TagGrid>
                                        </TypesBox>
                                    </DropdownGridRow>
                                    <DropdownGridRow>
                                        <DropdownHeader>Sort By</DropdownHeader>
                                        <SortByBox>
                                            <SortTag>A - Z</SortTag>
                                            <SortTag>Z - A</SortTag>
                                            <SortTag>Lowest No.</SortTag>
                                            <SortTag>Highest No.</SortTag>
                                            <SortTag style={{}}>Favorites</SortTag>
                                        </SortByBox>
                                    </DropdownGridRow>
                                    <DropdownGridRow>
                                        <ButtonsContainer>
                                            <ButtonOuter>
                                                <ResetBtn onClick={handleReset}>Reset</ResetBtn>
                                                <form onSubmit={handleSubmit} className="formButton">
                                                    <SearchBtn type="submit">Search</SearchBtn>
                                                </form>
                                            </ButtonOuter>
                                        </ButtonsContainer>
                                    </DropdownGridRow>
                                </DropdownOuter>
                            </DropdownContainer>
                        </>
                    )}
                </OverflowWrapper>
                <FilterOuter>
                    <NavbarEdge
                        className="navbarDropdown"
                        // style={{ marginTop: small || showDropdown ? "50px" : "10px" }}
                        src={navbarEdge}
                        onClick={() => setShowDropdown(!showDropdown)}
                    />
                </FilterOuter>
            </NavbarOuter>
            <LeftEdge src={leftEdge} height={lineHeight} top={lineTop} offset={lineOffset} />
            <RightEdge src={rightEdge} height={lineHeight} top={lineTop} offset={lineOffset} />
        </>
    );
};
export default Navbar;
