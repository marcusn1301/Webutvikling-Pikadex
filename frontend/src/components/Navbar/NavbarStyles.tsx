import styled from "styled-components";

// Media queries https://www.w3schools.com/css/css_rwd_mediaqueries.asp

interface Epic {
    height: any;
    transition: any;
}

//The wrapper for our Navbar
export const NavbarOuter = styled.div<Epic>`
    // Sets only animation when scrolling down
    transition: ${(props) => (props.transition ? "0.3s all" : "0s")};
    height: ${(props) => props.height};
    position: fixed;
    width: 100%;
    background-color: #bd1808;

    .pikachu {
        position: absolute;

        @media (max-width: 600px) {
            visibility: hidden;
        }
    }

    @media (max-width: 600px) {
        // Sets only animation when scrolling down
        transition: ${(props) => (props.transition ? "0.3s all" : "0s")};
        height: ${(props) => props.height};
        position: fixed;
        width: 100%;
        background-color: #bd1808;
    }
`;

//Wraps the content inside the navbar. Scroll on vertical overflow. Dynamic heighs.
export const OverflowWrapper = styled.div`
    position: relative;
    width: 100%;
    min-height: 172px;
    max-height: 85vh;

    -ms-overflow-style: none; /* for Internet Explorer, Edge */
    scrollbar-width: none; /* for Firefox */

    ::-webkit-scrollbar {
        display: none; /* for Chrome, Safari, and Opera */
    }
`;

//Wrapper for the logo and the textbox
export const LogoContainer = styled.div`
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 96%;
    margin: 15px 0 0 2%;
    height: 32px;
    background-color: rgba(234, 234, 234, 0.2);
    border-radius: 4px;
    color: white;
    font-size: 24px;
    @media (max-width: 600px) {
        display: flex;
        justify-content: space-between;
        align-items: center;
        width: 96%;
        margin: 15px 0 0 2%;
        height: 32px;
        background-color: rgba(234, 234, 234, 0.2);
        border-radius: 4px;
        color: white;
        font-size: 24px;
    }
`;

export const LogoTextbox = styled.div`
    /* transition: 0.5s all; */

    display: flex;
    justify-content: space-between;
    gap: 10px;
    @media (max-width: 600px) {
        display: flex;
        justify-content: space-between;
        gap: 10px;
    }
    &:hover {
        cursor: pointer;
        filter: invert(20%);
    }
`;

export const Logo = styled.img`
    width: 25px;
    margin-left: 8px;
    @media (max-width: 600px) {
        width: 25px;
        margin-left: 8px;
    }
`;

export const DownArrow = styled.img`
    transition: 0.5s all;
    width: 25px;
    position: relative;
    bottom: 4px;
    right: 5px;

    @media (max-width: 600px) {
        width: 25px;
        position: relative;
        bottom: 4px;
        right: 5px;
    }
`;

export const PokeBall = styled.img`
    width: 120px;
    position: absolute;
    right: 10px;
    top: 10px;
    @media (max-width: 600px) {
        width: 120px;
        position: absolute;
        right: 10px;
        top: 10px;
    }
`;

//The grid creates equally sized rows. Used by the search bar, pokeball, types-box, sort-by-box, and the
//buttons at the bottom of the navbar.
export const GridContainer = styled.div`
    display: grid;
    margin-top: 20px;
    padding: 0 20px 0 20px;
    grid-template-rows: repeat(1fr);
    row-gap: 20px;
    @media (max-width: 600px) {
        display: grid;
        margin-top: 20px;
        padding: 0 20px 0 20px;
        grid-template-rows: repeat(1fr);
        row-gap: 20px;
    }
`;

//Wrapper for the searchbar and pokeball button
export const SearchOuter = styled.div`
    display: grid;
    width: 65%;
    height: 40px;
    grid-template-columns: 1fr 4fr 2.5fr;
    grid-column-gap: 0px;
    margin: auto;
    @media (max-width: 950px) {
        grid-template-columns: 1fr 4fr 1fr;
    }

    .searchbar {
        outline: none;
        height: 38px;
        flex: 1;
        border: none;
        padding-left: 10px;
        font-size: 15px;
        width: 100%;
        margin-right: 20vw;
    }

    .searchContainer {
        display: inline-flex;
        flex: 1;
        border: 1px solid grey;
        background-color: #f4f4f4;
        border-radius: 6px;
        overflow: hidden;
    }

    .searchIcon {
        padding: 0.45rem;
        background-color: #f8f8f8;

        &:hover {
            cursor: pointer;
            background-color: rgb(200, 200, 200);
        }
        /* display: flex; */
        /* align-items: center; */
        /* background-color: black;
            -webkit-mask-image: url(icon.svg);
            mask-image: url(icon.svg); */
    }
    @media (max-width: 600px) {
        display: grid;
        width: 90%;
        height: 40px;
        grid-template-columns: 4fr 1fr;
        grid-column-gap: 5px;

        .filler {
            display: none;
        }

        .searchContainer {
            display: inline-flex;
            flex: 1;
            border: 1px solid grey;
            background-color: #f8f8f8;
            border-radius: 6px;
            overflow: hidden;
        }

        .searchbar {
            outline: none;
            /* padding: 0.5rem 0.5rem 0.5rem 0; */
            flex: 1;
            border: none;
            padding-left: 10px;
            font-size: 15px;
            width: 100%;
        }

        .searchIcon {
            padding: 0.45rem;
            background-color: #f8f8f8;
            /* display: flex; */
            /* align-items: center; */
            /* background-color: black;
            -webkit-mask-image: url(icon.svg);
            mask-image: url(icon.svg); */
        }
    }
    @media (min-width: 900px) {
        width: 55%;
    }
    @media (min-width: 1200px) {
        width: 45%;
    }
    @media (min-width: 1600px) {
        width: 55%;
    }
`;

export const PokeBallBtnContainer = styled.div`
    position: relative;
    width: 100%;
    display: flex;
    justify-content: center;
    @media (max-width: 600px) {
        position: relative;
        width: 100%;
        display: flex;
        justify-content: center;
    }
`;

export const PokeballBoxSurpriseMe = styled.div`
    color: white;
    border-radius: 6px;
    background-color: rgba(15, 10, 10, 0.7);
    width: 90%;
    display: flex;
    align-items: center;
    justify-content: center;
    gap: 5px;
    &:hover {
        cursor: pointer;
        background-color: rgba(15, 10, 10, 0.8);
    }
    @media (max-width: 950px) {
        background-color: transparent;
    }
`;

export const PokeballBoxSurpriseMeText = styled.div`
    @media (max-width: 950px) {
        display: none;
    }
`;

export const PokeBallBtn = styled.img`
    width: 40px;
    transition: 0.7s;
    @media (max-width: 600px) {
        width: 40px;
    }
    &:hover {
        cursor: pointer;
        transform: rotate(-360deg);
    }
`;

//Wrapper for the type-tags. Used for the carousel.
export const SelectedTags = styled.div`
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    width: 20%;
    column-gap: 20px;
    margin: auto;
    justify-content: center;
`;

//Wrapper for the type-tags. Used for the carousel.
export const TagGrid = styled.div`
    display: grid;
    height: 85%;
    margin: auto;
    grid-column-gap: 15px;
    grid-row-gap: 15px;

    //Carousel for the tags
    overflow-y: scroll;
    width: 90%;

    -ms-overflow-style: none; /* for Internet Explorer, Edge */
    scrollbar-width: none; /* for Firefox */
    overflow-y: scroll;

    ::-webkit-scrollbar {
        display: none; /* for Chrome, Safari, and Opera */
    }

    @media (max-width: 325px) {
        grid-template-columns: repeat(2, auto);
        height: min-content;
    }
    @media (min-width: 325px) {
        grid-template-columns: repeat(3, auto);
        height: min-content;
    }
    @media (min-width: 600px) {
        grid-template-columns: repeat(6, auto);
    }

    @media (min-width: 1050px) {
        resize: none;
    }
`;

interface TagI {
    backgroundColor: string;
}

//Type tag such as Water, Bug, Fire, etc.
export const Tag = styled.div<TagI>`
    color: #ffffff;
    transition: 0.1s;
    padding: 2px 3px 2px 3px;
    background-color: rgba(15, 10, 10, 0.7);
    font-weight: 500;
    letter-spacing: 1px;
    display: flex; //Center text within tag
    align-items: center;
    justify-content: space-evenly;
    border: 5px solid ${(props) => props.backgroundColor};
    border-radius: 20px;
    &:hover {
        cursor: pointer;
        background-color: rgba(15, 10, 10, 0.8);
    }
`;

interface FilterTagsI {
    background: string;
}

export const FilterTags = styled.div<TagI>`
    color: #ffffff;
    padding: 2px 3px 2px 3px;
    transition: 0.1s;
    background-color: rgba(15, 10, 10, 0.7);
    font-weight: 500;
    letter-spacing: 1px;
    display: flex; //Center text within tag
    align-items: center;
    justify-content: space-evenly;
    border: 5px solid ${(props) => props.backgroundColor};
    outline: 1px solid white;
    border-radius: 20px;
    &:hover {
        cursor: pointer;
        opacity: 0.85;
        background-color: rgba(15, 10, 10, 0.8);
    }
`;
//The bottom edge of the navbar
export const FilterOuter = styled.div`
    display: flex;
    justify-content: center;
    width: 100%;
`;

//Filter icon to the left of the text
export const FilterIcon = styled.div``;

//"Filter your pokemon"
export const FilterText = styled.div``;

//Adjust width based on screen size
export const DropdownContainer = styled.div`
    @media (min-width: 600px) {
        width: 90%;
        margin: auto;
    }
    @media (min-width: 1050px) {
        width: 80%;
        margin: auto;
    }
    @media (min-width: 1500px) {
        width: 70%;
        margin: auto;
    }
    @media (min-width: 1900px) {
        width: 60%;
        margin: auto;
    }
`;

//The dropdown appears when "Filter your pokemon" is clicked. The grid creates equally sized rows.
export const DropdownOuter = styled.div`
    transition: 0.5s all;
    position: relative;
    background-color: #bd1808;
    width: 100%;
    height: 500px;
    z-index: 999;
    margin-top: 15px;
    padding-top: 15px;
    text-align: center;
    border-top: 1px solid rgba(255, 255, 255, 0.2);
    display: grid;
    grid-row-gap: 50px;

    @media (max-width: 600px) {
        grid-template-rows: 3fr 3fr 1fr;
    }

    @media (min-width: 600px) {
        grid-template-columns: 2fr 1fr;
        height: 300px;
    }
`;

//Wrapper for each row-section in the dropdown grid
export const DropdownGridRow = styled.div`
    width: 100%;

    @media (max-width: 600px) {
        &:nth-child(3) {
            padding-bottom: 10px;
        }
    }

    //The buttons at the bottom of the navbar takes up two colunmns
    @media (min-width: 600px) {
        &:nth-child(3) {
            grid-column: span 2;
            width: 90%;
            display: flex;
            align-items: center;
            margin: auto;
        }
    }
`;

//Header for "Types" and "Sort by"
export const DropdownHeader = styled.div`
    color: #ffffff;
    font-size: 1.2rem;
`;

//Box and grid for the types-section. Parent of the TagOuter containing the carousel.
export const TypesBox = styled.div`
    background-color: rgba(255, 255, 255, 0.2);
    display: grid;
    align-items: center;
    height: 100%;
    border-radius: 10px;
    width: 95%;
    margin: auto;
    z-index: 999999;
`;

//Box and grid for the sort-by-section
export const SortByBox = styled.div`
    height: 100%;
    background-color: rgba(255, 255, 255, 0.2);
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(auto, auto));
    border-radius: 10px;
    width: 95%;
    align-items: center;
    margin: auto;
    z-index: 99999999999;
`;

//"Sort by"-tags such as "A - Z"
export const SortTag = styled.div`
    background-color: rgba(15, 10, 10, 0.7);
    color: #ffffff;
    border-radius: 10px;
    font-size: 1.2rem;
    width: 80%;
    height: 40px;
    display: grid;
    align-items: center;
    margin: auto;

    //Last tag takes up width of two columns
    &:nth-child(5) {
        border-spacing: 10px;
        grid-column: span 2;
        width: 90%;
    }

    font-size: 0.9rem;
    padding: 0 5px 0 5px;
    @media screen and (max-width: 320px) {
        font-size: 0.9rem;
        padding: 0 5px 0 5px;
    }
    &:hover {
        cursor: pointer;
        background-color: rgba(15, 10, 10, 0.8);
    }
`;

export const ButtonsContainer = styled.div`
    width: 100%;
    display: flex;
    justify-content: center;

    .formButton {
        width: 100%;
    }
`;

export const ButtonOuter = styled.div`
    display: grid;
    grid-template-columns: 1fr 1fr;
    @media screen and (max-width: 600px) {
        width: 90%;
    }
    @media screen and (min-width: 600px) {
        width: 40%;
    }
`;

export const ResetBtn = styled.div`
    background-color: rgba(15, 10, 10, 0.7);
    color: #ffffff;
    padding: 5px 0 5px 0;
    width: 70%;
    border-radius: 10px;
    font-size: 1.2rem;
    margin: auto;

    @media screen and (max-width: 320px) {
        font-size: 1rem;
    }
    &:hover {
        cursor: pointer;
        background-color: rgba(15, 10, 10, 0.8);
    }
`;

export const SearchBtn = styled.button`
    background-color: rgba(15, 10, 10, 0.7);
    color: #ffffff;
    width: 100%;
    height: 100%;
    border: none;
    border-radius: 10px;
    font-size: 1.2rem;

    @media screen and (max-width: 320px) {
        font-size: 1rem;
    }
    &:hover {
        cursor: pointer;
        background-color: rgba(15, 10, 10, 0.8);
    }
`;

export const PokemonContainer = styled.div`
    width: 25%;
    display: grid;
    position: absolute;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    height: 50px;
    right: 5px;
    overflow: hidden;
    top: 122px;
    column-gap: 0px;
    align-items: baseline;

    img {
        position: relative;
        bottom: 0px;
        width: 55px;
    }

    @media (max-width: 600px) {
        visibility: hidden;
    }

    .pikachu-small {
        transform: translateY(6px);
    }
    .bulbasaur {
        transform: translateY(9px);
    }
    .charmander {
        transform: translateY(5px);
    }
    .squirtle {
        transform: translateY(6px);
    }
`;

export const NavbarEdge = styled.img`
    position: absolute;
    bottom: -22px;
    z-index: 999;
    width: 270px;

    &:hover {
        cursor: pointer;
        filter: invert(7.5%);
    }

    @media (min-width: 600px) {
        width: 300px;
    }
`;

// interfaces for side lines
interface LeftEdgeI {
    height: string;
    top: string;
    offset: string;
}

interface RightEdgeI {
    height: string;
    top: string;
    offset: string;
}

export const LeftEdge = styled.img<LeftEdgeI>`
    transition: 0.2s all;
    position: fixed;
    left: ${(props) => "-" + props.offset};
    top: ${(props) => props.top};
    height: ${(props) => props.height};
    z-index: -1;

    @media (max-height: 680px) {
        height: 64vh;
    }

    @media (max-height: 550px) {
        height: 55vh;
    }

    @media (max-height: 450px) {
        height: 45vh;
    }

    @media (max-height: 400px) {
        height: 40vh;
    }
`;

export const RightEdge = styled.img<RightEdgeI>`
    transition: 0.2s all;
    position: fixed;
    right: ${(props) => "-" + props.offset};
    transform: translateX(0px);
    top: ${(props) => props.top};
    height: ${(props) => props.height};
    z-index: -1;

    @media (max-height: 680px) {
        height: 64vh;
    }

    @media (max-height: 550px) {
        height: 55vh;
    }

    @media (max-height: 450px) {
        height: 45vh;
    }

    @media (max-height: 400px) {
        height: 40vh;
    }
`;
