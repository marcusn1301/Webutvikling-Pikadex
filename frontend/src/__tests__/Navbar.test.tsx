import React from 'react';
import {render, screen} from '@testing-library/react';
import renderer from "react-test-renderer";

import Navbar from '../components/Navbar/Navbar';

test('Snapshot test for Navbar', () => {
    const container = renderer.create(<Navbar />);
    expect(container).toMatchSnapshot();
});

