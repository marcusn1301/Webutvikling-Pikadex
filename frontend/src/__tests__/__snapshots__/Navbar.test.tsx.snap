// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for Navbar 1`] = `
Array [
  <div
    className="sc-bcXHqe Gybjo"
    height="172px"
  >
    <div
      className="sc-gswNZR egMUqU"
      style={
        Object {
          "minHeight": "173px",
          "overflowY": "hidden",
        }
      }
    >
      <div
        className="sc-dkrFOg icdcLv"
      >
        <div
          className="sc-hLBbgP kFvYYM"
        >
          <img
            className="sc-eDvSVe dIDjLj"
            src="Logo.svg"
          />
          <div>
            Pikadex
          </div>
        </div>
      </div>
      <img
        alt="pikachu"
        className="pikachu"
        src="pikachu.svg"
        style={
          Object {
            "display": "block",
          }
        }
      />
      <div
        className="sc-iJnaPW cuGxOM"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <img
          alt="pikachu"
          className="pikachu-small"
          src="pikachu.svg"
        />
        <img
          alt="bulbasaur"
          className="bulbasaur"
          src="bulbasaur.svg"
        />
        <img
          alt="charizard"
          className="charmander"
          src="charmander.svg"
        />
        <img
          alt="squirtle"
          className="squirtle"
          src="squirtle.svg"
        />
      </div>
      <img
        className="sc-gKPRtg dwhDdR"
        src="Pokeball.svg"
      />
      <div
        className="sc-iBYQkv jzCrYC"
      >
        <div
          className="sc-ftTHYK bcWsZC"
        >
          <div
            className="filler"
          />
          <div
            className="searchContainer"
          >
            <form
              onSubmit={[Function]}
            >
              <input
                className="searchbar"
                onChange={[Function]}
                placeholder="Search by name or index"
                type="text"
              />
            </form>
            <svg
              ariaHidden="true"
              className="searchIcon"
              fill="currentColor"
              height={25}
              onClick={[Function]}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width={25}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M10.5 3.75a6.75 6.75 0 100 13.5 6.75 6.75 0 000-13.5zM2.25 10.5a8.25 8.25 0 1114.59 5.28l4.69 4.69a.75.75 0 11-1.06 1.06l-4.69-4.69A8.25 8.25 0 012.25 10.5z"
                fillRule="evenodd"
              />
            </svg>
          </div>
          <div
            className="sc-pyfCe hBdHyt"
            onClick={[Function]}
          >
            <div
              className="sc-jrcTuL gXuwBj"
            >
              <div
                className="sc-kDvujY bBbyqe"
              >
                Surprise me!
              </div>
              <img
                className="sc-ipEyDJ iGiXAa"
                src="pokeball_icon.svg"
              />
            </div>
          </div>
        </div>
        <div
          className="sc-csuSiG eHinTQ"
        />
      </div>
    </div>
    <div
      className="sc-hBxehG YAnpd"
    >
      <img
        className="sc-gikAfH ijcsKr navbarDropdown"
        onClick={[Function]}
        src="navbarEdge.svg"
      />
    </div>
  </div>,
  <img
    className="sc-ezOQGI dOxhFP"
    height="70vh"
    offset="1px"
    src="leftEdge.svg"
  />,
  <img
    className="sc-bYMpWt ckrZjW"
    height="70vh"
    offset="1px"
    src="rightEdge.svg"
  />,
]
`;
